### ISSUE TITLE
Support of allowed_client_headers_on_success in extAuth http SecurityPolicy

### ISSUE THREAD

**Issue by gerrieg** [envoyproxy/gateway#3787]
We need to set a cookie in our ext auth service, for this we need [allowed_client_headers_on_success](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto#envoy-v3-api-field-extensions-filters-http-ext-authz-v3-authorizationresponse-allowed-client-headers-on-success) 

Currently we use EnvoyPatchPolicy to add the configuration, but native support would be great

Thank you

**Comment by svrakitin** [comment]
Would really like to get support for `allowed_client_headers` and `allowed_client_headers_on_success` as it currently forces us to use `EnvoyPatchPolicy`. I also see that the code handling headers like `headersToBackend` assumes a list of headers vs. a `StringMatcher`, which makes it impossible to handle header prefixes.

I'd like to contribute, but it seems like it needs a much bigger cleanup and someone with more context. :)

**Comment by mou** [comment]
We also have same requirements. Our ext_authz service is responsible for ratelimiting decisions, so the rate-limit headers it emits should be passed downstream. Envoy proxy allows to implement it, and we want Envoy Gateway to support this capability as well.

**Comment by shnipov** [comment]
Yeah, this will remove a lot of pain for us too. Just clean config, no more dirty workarounds.

**Comment by Pybsama** [comment]
Hi, I'd like to work on this. Since it requires a change under `/api`, I'd like
to agree on the API shape before starting the implementation.

For the original success-response use case, I propose adding this optional
field under `HTTPExtAuthService`:

```go
HeadersToClientOnSuccess []string `json:"headersToClientOnSuccess,omitempty"`
```

It would map to Envoy's `allowed_client_headers_on_success`. Each configured
header would use the same exact, case-insensitive matching behavior as the
existing `headersToBackend` field. When omitted or empty, Envoy Gateway would
leave `allowed_client_headers_on_success` unset and preserve the current
behavior.

I propose keeping this issue focused on successful authorization responses.
Support for denied-response headers, prefix/suffix/regex matchers, or changing
the existing `headersToBackend []string` API would remain separate follow-up
work.

Would this field name and exact-list behavior be acceptable? Would you prefer
an API-only PR before the IR/xDS implementation?

**Comment by arkodg** [comment]
go for it @Pybsama !, all changes in 1 PR is fine for this feature
